### DISCUSSION TITLE
QSelect: Can't scroll to before-options slot

### DISCUSSION THREAD

**Discussion by tohagan** [quasarframework/quasar#13597] [General - Components / Directives / etc] [answered]
### What happened?

I've developing a QPhoneInput component to validate and format international phone numbers. 
It uses a QSelect to display international dialing codes with flags that has a Search filter (QInput) in the `before-options` slot and duplicated in the `no-option` slot in case the options list becomes empty (due to #6839 )

When the user taps the QSelect to show the country list options, I set focus to the Search filter and am now *trying* to find a way to set virtual scroller to the `before-options` slot but can't a way to do so (`scrollTo(0)` scrolls to the 1st item, not the `before-options` slot).  

See attempt at [CountrySelection.vue L163](https://github.com/tohagan/quasar-q-phone-input/blob/9ce352fe621a1ed9b0fa523fb14c60a18aa690bc/src/components/phone/CountrySelection.vue#L163
)

### What did you expect to happen?

**Plan A: Fix**. Support some way (or help me find one) to set the virtual scroller to the top to make the before-options slot visible when the QSelect is shown. 

**Plan B: Feature**. Add a boolean property so that the before-options slot is always visible (not part of the virtual scroller) so we can remove duplication of the filter control and the need to reset scrolling. I think this slot design makes most sense for the common case of filtering the options.  I think this is the expected behaviour as this issue also suggests #6839 .  This new boolean property would need to be false by default for backward compatibility. 

### Reproduction URL

https://stackblitz.com/edit/quasar-q-phone-input

### How to reproduce?

1. Open the StackBlitz.com link and run `quasar dev`
2. Tap the country prefix flag to select a new country
3. You're forced to scroll up to find the search box 

Source for StackBlitz.com is also at https://github.com/tohagan/quasar-q-phone-input

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Components (quasar)

### Platforms/Browsers

Chrome

**Comment by metalsadman** [comment, marked as answer]
because before-slot isn't part of the options list (not a child of virtual scroll div). you can make the slot sticky using css https://codepen.io/metalsadman/pen/eYgbMVV

**Reply by tohagan** [reply]
Works perfectly. Thank you!
